### PR TITLE
Export ClipImageTransform

### PR DESCRIPTION
**Pull request by lucylq**
#### Context
What is the purpose of this PR? Is it to
- [x] add a new feature
- [ ] fix a bug
- [ ] update tests and/or documentation
- [ ] other (please add here)

Please link to any issues this PR addresses.

#### Changelog
- Add torchtune/models/clip/inference/_transforms.py, a version of CLIPImageTransform that can be exported. Update transforms tests with a comparison between the two CLIPImageTransforms. 
- Refactor target_size calculations into function `get_resize_dimensions`, add tests.
- Change tile_crop from a function to an nn.Module for module swapping, update callsites.

Note: we cannot add a test for the exported model, as source transforms are required on the nn.Modules to successfully export it. 

#### Test plan
Please make sure to do each of the following if applicable to your PR. (If you're not sure about any one of these just ask and we will happily help. We also have a [contributing page](../CONTRIBUTING.md) for some guidance on contributing.)

- [x] run pre-commit hooks and linters (make sure you've first installed via `pre-commit install`)
- [x] add [unit tests](../tests/torchtune) for any new functionality
- [x] update [docstrings](../docs/source) for any new or updated methods or classes
- [x] run unit tests via `pytest tests`
- [ ] run recipe tests via `pytest tests -m integration_test`
- [ ] manually run any new or modified recipes with sufficient proof of correctness
- [ ] include relevant commands and any other artifacts in this summary (pastes of loss curves, eval results, etc.)

### CODE REVIEW

**Review by felipemello1 — COMMENTED** [review]
LGTM, I will leave the final word to @pbontrager. Maybe it would be nice to add a small docstring at the top of "torchtune/models/clip/inference/_transforms.py" explaining that this is for executorch only. But I am fine with leaving it as it is.

File: tests/torchtune/models/clip/test_clip_image_transform.py
Comment (lucylq) [reply tests/torchtune/models/clip/test_clip_image_transform.py]: Ah, good catch. Updated

Comment (lucylq) [reply tests/torchtune/models/clip/test_clip_image_transform.py]: Also updated - @tarun292 found we can swap out functionals as long as we 'name' them inside __init__(), so leaving torch.nn.functional.pad, and tile_crop as is.

**Comment by lucylq** [comment]
> LGTM, I will leave the final word to @pbontrager. Maybe it would be nice to add a small docstring at the top of "torchtune/models/clip/inference/_transforms.py" explaining that this is for executorch only. But I am fine with leaving it as it is.

Thanks for reviewing! Added a note that being able to export is useful for ExecuTorch/AOTI.

File: torchtune/modules/transforms/vision_utils/get_resize_dimensions.py
Comment (pbontrager) [inline torchtune/modules/transforms/vision_utils/get_resize_dimensions.py:20]: I think this could be fleshed out a bit more. The key thing this function does is computes what size an image would be if it was resized to be inscribed within a target size. So it's upscaled or downscaled such that one size equals the target size and the second side is less than or equal to the target size. I think some description describing this would help make the function easier to discover and use.

File: torchtune/modules/transforms/vision_utils/get_resize_dimensions.py
Comment (pbontrager) [inline torchtune/modules/transforms/vision_utils/get_resize_dimensions.py:16]: This needs to be added to the docs RST

File: torchtune/modules/transforms/vision_utils/get_resize_dimensions.py
Comment (pbontrager) [inline torchtune/modules/transforms/vision_utils/get_resize_dimensions.py:41]: @felipemello1 I find it counterintuitive that this doesn't cause a downsize to (225, 600). max_upscaling_size is a bit weird to have here at all because you could just not provide a target that's too big, but I understand it's convenient.

**Review by pbontrager — COMMENTED** [review]
This is looking nice and very clean. I really like the docstring for the new transform as well. I left some comments around naming and descriptions but mostly nits. I think the get_resize_dimensions should be added to the docs, but I'm fine with leaving the inference transform out of the docs for now as we figure out our story there.

**Review by pbontrager — COMMENTED** [review]
This is looking nice and very clean. I really like the docstring for the new transform as well. I left some comments around naming and descriptions but mostly nits. I think the get_resize_dimensions should be added to the docs, but I'm fine with leaving the inference transform out of the docs for now as we figure out our story there.

File: torchtune/modules/transforms/vision_utils/get_resize_dimensions.py
Comment (lucylq) [reply torchtune/modules/transforms/vision_utils/get_resize_dimensions.py:26]: Sounds good - I'll change this in the other transforms as well so it's consistent.

File: torchtune/modules/transforms/vision_utils/get_resize_dimensions.py
Comment (lucylq) [reply torchtune/modules/transforms/vision_utils/get_resize_dimensions.py]: sounds good - updated

File: torchtune/modules/transforms/vision_utils/get_resize_dimensions.py
Comment (lucylq) [reply torchtune/modules/transforms/vision_utils/get_resize_dimensions.py]: good catch - updated

File: torchtune/modules/transforms/vision_utils/get_resize_dimensions.py
Comment (lucylq) [reply torchtune/modules/transforms/vision_utils/get_resize_dimensions.py]: thanks!

File: torchtune/modules/transforms/vision_utils/get_resize_dimensions.py
Comment (lucylq) [reply torchtune/modules/transforms/vision_utils/get_resize_dimensions.py]: thanks, updated

**Review by pbontrager — APPROVED** [review]
Thanks for the hard work! I'll approve this now, but the test_get_resize_dimensions test should be renamed before you land it.

**Pull request merged**
